Sources/XcodeProj/Workspace/XCWorkspace.swift

```diff
@@ -53,6 +53,15 @@ public final class XCWorkspace: Writable, Equatable {
     public func write(path: Path, override: Bool = true) throws {
         let dataPath = path + "contents.xcworkspacedata"
         if override, dataPath.exists {
+            if let existingContent: String = try? dataPath.read(),
```

Comment:
IMHO guard is more explicit if you want to do early exit (as in this case)

Reply:
Yes, but personally I only like to use guards where it reads well.

In this case we're looking for identical data, which suggests `==` and not `!=`.

There's something to be said about the simplicity of an `if`, and not having to backtrack the negative boolean checks that read: "guard if not the case otherwise that" (as opposed to "if case then").

IMHO early escapes should be easy to comprehend, but the `guard` statement complicates them more often than not.

Other than it being a signpost for an early escape, are there other advantages to using a `guard` in this case that I'm unaware of?

If you feel particularly strong on his point I'm fine going with the `guard` asw 🙂